Dockerfile

```diff
@@ -1,32 +1,32 @@
+FROM python:3.7-slim-buster AS cache
+
+COPY . /opt/CTFd
+RUN find /opt/CTFd -not -name "requirements.txt" -type f -delete
```

Comment:
# delete everything **except** requirements.txt, so when actually copied to the second stage, docker will get the same hashes even if the plugin source has changed, as long as the requirements.txt doesn't change. the packages still remains in the "main image", if I understanded your comment correctly.